quarkus-workshop-super-heroes/docs/src/docs/asciidoc/5a-contract-testing/happy-path-consumer-tests.adoc

```diff
@@ -0,0 +1,166 @@
+----
+
+The `%test` scopes the updated property to only apply in the test mode.
+
+We will be using the https://docs.pact.io/implementation_guides/jvm/consumer/junit5[junit 5 pact libraries].
+
+Create a class called `FightResourceConsumerTest.java`.
+
+[source,java]
+----
+package io.quarkus.workshop.superheroes.fight;
+
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.PactSpecVersion;
+import au.com.dius.pact.core.model.V4Pact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.workshop.superheroes.fight.client.DefaultTestVillain;
+import org.hamcrest.core.Is;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import java.util.Map;
+
+import static au.com.dius.pact.consumer.dsl.LambdaDsl.newJsonBody;
+import static io.quarkus.workshop.superheroes.fight.client.DefaultTestHero.DEFAULT_HERO_LEVEL;
+import static io.quarkus.workshop.superheroes.fight.client.DefaultTestHero.DEFAULT_HERO_NAME;
+import static io.quarkus.workshop.superheroes.fight.client.DefaultTestHero.DEFAULT_HERO_PICTURE;
+import static io.restassured.RestAssured.given;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.Response.Status.OK;
+
+@QuarkusTest
+@ExtendWith(PactConsumerTestExt.class)
+@PactTestFor(
+    providerName = "rest-heroes",
+    // Hard-code the Pact MockServer to what we defined in application.properties
+    port = "8093"
+)
+public class FightResourceConsumerTest {
+
+    protected static final String HERO_API_BASE_URI = "/api/heroes";
+    protected static final String HERO_RANDOM_URI = HERO_API_BASE_URI + "/random";
+
+    @Pact(consumer = "rest-fights")
+    public V4Pact randomHeroFoundPact(PactDslWithProvider builder) {
+        return builder
+            .uponReceiving("A request for a random hero")
+            .path(HERO_RANDOM_URI)
+            .method(HttpMethod.GET)
+            .headers(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
+            .willRespondWith()
+            .status(Response.Status.OK.getStatusCode())
+            .headers(Map.of(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON))
+            .body(newJsonBody(body ->
+                    body
+                        .stringType("name", DEFAULT_HERO_NAME)
+                        .integerType("level", DEFAULT_HERO_LEVEL)
+                        .stringType("picture", DEFAULT_HERO_PICTURE)
+                ).build()
+            )
+            .toPact(V4Pact.class);
+    }
+}
+----
```

Comment:
Maybe do some indents so it makes it easier to read?

Also I think the static imports from `DefaultTestHero` make it more confusing? Maybe remove the static imports and make the calls explicitly?

```java
    @Pact(consumer = "rest-fights")
    public V4Pact randomHeroFoundPact(PactDslWithProvider builder) {
        return builder
            .uponReceiving("A request for a random hero")
                .path(HERO_RANDOM_URI)
                .method(HttpMethod.GET)
                .headers(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
            .willRespondWith()
                .status(Response.Status.OK.getStatusCode())
                .headers(Map.of(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON))
                .body(newJsonBody(body ->
                        body
                            .stringType("name", DefaultTestHero.DEFAULT_HERO_NAME)
                            .integerType("level", DefaultTestHero.DEFAULT_HERO_LEVEL)
                            .stringType("picture", DefaultTestHero.DEFAULT_HERO_PICTURE)
                    ).build()
                )
            .toPact(V4Pact.class);
    }
```

Reply:
I think indents will be a challenge without adding some formatting meta-data ... and to me, with the methods not aligned, it looks a bit like there's an extra set of brackets/level of nesting that isn't actually there in the code. It's a bit of a weakness of the Pact DSL, it would be nice if that nesting *was* there. 
I've swapped out the static imports. I agree they were probably concise, but more confusing.